pkgs/stdenv/darwin/default.nix

```diff
@@ -503,7 +506,6 @@ in rec {
     targetPlatform = localSystem;
 
     preHook = commonPreHook + ''
-      export NIX_COREFOUNDATION_RPATH=${pkgs.darwin.CF}/Library/Frameworks
```

Comment:
The main reason this is set directly in the stdenv is to ensure that any usage of frameworks takes priority. The goal here is to link against the pure version as much as possible, but system frameworks also link against the CoreFoundation framework in which case also linking against the pure build causes problems.

Reply:
I think that the ordering might be enforced by `extraBuildInputs`, but just in case there is a simple bash logic in corefoundation-setup-hook.sh to avoid overriding `NIX_COREFOUNDATION_RPATH`.